frappe/api.py

```diff
@@ -108,25 +109,39 @@ def handle():
 
 			elif doctype:
 				if frappe.local.request.method == "GET":
-					if frappe.local.form_dict.get('fields'):
-						frappe.local.form_dict['fields'] = json.loads(frappe.local.form_dict['fields'])
-					frappe.local.form_dict.setdefault('limit_page_length', 20)
-					frappe.local.response.update({
-						"data": frappe.call(
-							frappe.client.get_list,
-							doctype,
-							**frappe.local.form_dict
-						)
-					})
+					# set fields for frappe.get_list
+					if frappe.local.form_dict.get("fields"):
+						frappe.local.form_dict["fields"] = json.loads(frappe.local.form_dict["fields"])
+
+					# set limit of records for frappe.get_list
+					frappe.local.form_dict.setdefault(
+						"limit_page_length",
+						frappe.local.form_dict.limit or frappe.local.form_dict.limit_page_length or 20,
+					)
+
+					# convert strings to native types
+					frappe.local.form_dict.update(
+						{x: sbool(y) for x, y in frappe.local.form_dict.items()}
```

Comment:
this would convert in all these cases

![image](https://user-images.githubusercontent.com/28212972/119487423-fb9b9480-bd76-11eb-9b62-dda6068bd695.png)


This would be applicable to all parameters regardless of what was expected

One idea that I have is that can we keep the util and apply it to a parameter if we are expecting a boolean value?

Reply:
Given the scope of its usage at this point, this becomes a problem when you'd have a field named `y`,`n`, `true`, `false` and `order_by` that field, or have the same values for a `document name` that `parent` parameter would accept out of all those that Frappe REST allows.

OTOH, since only debug and as_dict can have boolean values, might as well just convert them 😄 . Nice catch